pkg/controllers/qat/controller.go

```diff
@@ -164,6 +183,75 @@ func (c *controller) UpdateStatus(rawObj client.Object, ds *apps.DaemonSet, node
 	return updated, nil
 }
 
+func removeVolume(volumes []v1.Volume, name string) []v1.Volume {
+	newVolumes := []v1.Volume{}
+
+	for _, volume := range volumes {
+		if volume.Name != name {
+			newVolumes = append(newVolumes, volume)
+		}
+	}
+
+	return newVolumes
+}
+
+func setInitContainer(dsSpec *v1.PodSpec, dpSpec devicepluginv1.QatDevicePluginSpec) {
+	yes := true
+
+	qatDeviceDriver := map[string]string{
+		"dh895xccvf": "0434 0435",
+		"c3xxxvf":    "19e2",
+		"c6xxvf":     "37c8",
+		"d15xxvf":    "6f54",
+		"4xxxvf":     "4940",
+		"c4xxxvf":    "18a0",
+	}
+
+	enablingPfPciIDs := ""
+	for _, v := range dpSpec.KernelVfDrivers {
+		enablingPfPciIDs = enablingPfPciIDs + qatDeviceDriver[string(v)] + " "
+	}
```

Comment:
```suggestion
	enablingPfPciIDs := make([]string, 0, len(qatDeviceDriver))

	for _, v := range dpSpec.KernelVfDrivers {
		if _, ok := qatDeviceDriver[v]; ok {
			enablingPfPciIDs = append(enablingPfPciIDs, v)
		}
	}
```

Reply:
Umm, is it to check whether the value of input is valid or not?

Reply:
my suggestion was mainly related to building an array and then using `strings.Join()` later. to avoid casting you can built the map with `KernelVfDriver`type